ministryofjustice/analytics-platform-ops
# Create EBS Snapshots

## What
Trello: https://trello.com/c/DAeTcJoT

Create ebs snapshots of etcd volumes

Frequency: daily

## How to review

1. `terraform plan -target=module.kubernetes_etcd_ebs_snapshot` __Done__
2. `terraform apply -target=module.kubernetes_etcd_ebs_snapshot` __Done__
3. Login to aws [console](https://us-east-1.signin.aws.amazon.com/oauth?SignatureVersion=4&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJMOATPLHVSJ563XQ&X-Amz-Date=2017-12-06T09%3A54%3A40.100Z&X-Amz-Signature=bd5db0db4d4c35fb5ac8dbe3c6ff85282305f21aa4f8af0fdad2237f7cbda4a9&X-Amz-SignedHeaders=host&client_id=arn%3Aaws%3Aiam%3A%3A015428540659%3Auser%2Fhomepage&redirect_uri=https%3A%2F%2Fconsole.aws.amazon.com%2Fconsole%2Fhome%3Fstate%3DhashArgs%2523%26isauthcode%3Dtrue&response_type=code&state=hashArgs%23)
4. View Cloud watch [Logs](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logStream:group=/aws/lambda/create_etcd_ebs_snapshot;streamFilter=typeLogStreamPrefix)
5. Sanity check Function by following [README](https://github.com/ministryofjustice/analytics-platform-ops/tree/ebs-backups/infra/terraform/global/assets/create_etcd_ebs_snapshot)
